server/channels/store/localcachelayer/content_flagging.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```

Comment:
I like the enthusiasm here but not sure if cache is necessary, did you run any tests for this? 😅